raiden-ts/src/services/epics.ts

```diff
@@ -903,3 +749,297 @@ export const msMonitorNewBPEpic = (
+            tokenNetwork: tokenNetwork,
+            serviceAddress: iou.receiver,
+          });
+        }
+      }
+
+      if (error) {
+        if (isNoRouteFoundError(error)) {
+          throw new RaidenError(ErrorCodes.PFS_NO_ROUTES_BETWEEN_NODES);
+        } else {
+          throw new RaidenError(ErrorCodes.PFS_ERROR_RESPONSE, {
+            errorCode: error.error_code,
+            errors: error.errors,
+          });
+        }
+      }
+
+      const filteredPaths = filterPaths(action, deps, latest, paths);
+
+      if (filteredPaths.length) {
+        yield pathFind.success({ paths: filteredPaths }, action.meta);
+      } else {
+        throw new RaidenError(ErrorCodes.PFS_NO_ROUTES_FOUND);
+      }
+    })(),
+  );
+}
+
+function validateRouteTargetAndEventuallyThrow(action: pathFind.request, latest: Latest): void {
+  const { tokenNetwork, target } = action.meta;
+  const { state, presences } = latest;
+
+  if (!Object.values(state.tokens).includes(tokenNetwork))
+    throw new RaidenError(ErrorCodes.PFS_UNKNOWN_TOKEN_NETWORK, { tokenNetwork });
+
+  if (!presences[target].payload.available)
+    throw new RaidenError(ErrorCodes.PFS_TARGET_OFFLINE, { target });
+
+  if (presences[target].payload.caps?.[Capabilities.NO_RECEIVE])
+    throw new RaidenError(ErrorCodes.PFS_TARGET_NO_RECEIVE, { target });
+}
+
+function pfsIsDisabled(action: pathFind.request, latest: Latest): boolean {
+  const disabledByAction = action.payload.pfs === null;
+  const disabledByConfig = !action.payload.pfs && latest.config.pfs === null;
+  return disabledByAction || disabledByConfig;
+}
+
+function getRouteFromPfs$(
+  action: pathFind.request,
+  deps: RaidenEpicDeps,
+  latest: Latest,
+): Observable<Route> {
+  const { tokenNetwork, target, value } = action.meta;
+
+  return getPsfInfo$(action.payload.pfs, latest.config.pfs, deps).pipe(
+    mergeMap((pfs) => getIouForPfs(pfs, tokenNetwork, deps)),
+    mergeMap(({ pfs, iou }) =>
+      requestPfs$(
+        pfs,
+        iou,
+        tokenNetwork,
+        deps.address,
+        target,
+        UInt(32).encode(value),
+        latest.config.httpTimeout,
+      ),
+    ),
+    map(({ pfsResponse, responseText, iou }) =>
+      parsePfsResponse(pfsResponse, responseText, iou, latest.config.pfsSafetyMargin),
+    ),
+  );
+}
+
+function filterPaths(
+  action: pathFind.request,
+  deps: RaidenEpicDeps,
+  latest: Latest,
+  paths: Paths | undefined,
+): Paths {
+  const { address, log } = deps;
+  const filteredPaths: Paths = [];
+  const invalidatedRecipients = new Set<Address>();
+
+  if (paths) {
+    for (const { path, fee } of paths) {
+      const cleanPath = getCleanPath(path, address);
+      const recipient = cleanPath[0];
+      let shouldSelectPath = false;
+      let reasonToNotSelect = '';
+
+      if (invalidatedRecipients.has(recipient)) continue;
+      if (filteredPaths.length === 0) {
+        shouldSelectPath = directTransferIsPossible(action, latest, recipient);
```

Comment:
In this case, if `directTransferIsPossible` returns falsy, you don't get a `reasonToNotSelect`

Reply:
Oh f***. This is pretty bad on a new level. :see_no_evil:

Reply:
Okay I reverted the introduction of this utility function. The `channelCanRoute` function is not easy to handle and an awkward approach to me. Anyways I'm using it directly again and somehow handle its result. This is now getting the reason again.

Reply:
:heavy_check_mark: